marketplace/commercetools-sku/src/index.js

```diff
@@ -0,0 +1,176 @@
+    installation: { apiEndpoint, authApiEndpoint, projectKey, clientId, clientSecret }
+  }
+}) {
+  const authMiddleware = createAuthMiddlewareForClientCredentialsFlow({
+    host: authApiEndpoint,
+    projectKey: projectKey,
+    credentials: {
+      clientId: clientId,
+      clientSecret: clientSecret
+    }
+  });
+
+  const httpMiddleware = createHttpMiddleware({
+    host: apiEndpoint
+  });
+
+  const queueMiddleware = createQueueMiddleware({
+    concurrency: 5
+  });
+
+  return createClient({
+    middlewares: [authMiddleware, httpMiddleware, queueMiddleware]
+  });
+}
+
+const fetchProductPreviews = async function fetchProductPreviews(skus, config) {
+  if (!skus.length) {
+    return [];
+  }
+
+  const client = makeCommerceToolsClient({ parameters: { installation: config } });
+  const requestBuilder = createRequestBuilder({ projectKey: config.projectKey });
+  const uri = requestBuilder.productProjectionsSearch
+    .parse({
+      filter: [`variants.sku:${skus.map(sku => `"${sku}"`).join(',')}`]
+    })
+    .build();
+  const response = await client.execute({ uri, method: 'GET' });
+  if (response.statusCode === 200) {
+    const products = response.body.results.map(dataTransformer(config));
+    return products;
+  }
+
+  return dataTransformer({}); // return empty product
+};
+
+async function renderDialog(sdk) {
+  const { projectKey, locale } = sdk.parameters.installation;
+
+  const client = makeCommerceToolsClient(sdk);
+
+  const ID = 'dialog-root';
+  const container = document.createElement('div');
+  container.id = ID;
+  container.style.display = 'flex';
+  container.style.flexDirection = 'column';
+  document.body.appendChild(container);
+
+  renderSkuPicker(ID, {
+    sdk,
+    fetchProductPreviews,
+    fetchProducts: async (search, pagination) => {
+      const PER_PAGE = 20;
+      const requestBuilder = createRequestBuilder({ projectKey });
+      const uri = requestBuilder.productProjectionsSearch
+        .parse({
+          ...(!!search.length && {
+            text: {
+              language: locale,
+              value: search
+            }
+          }),
+          page: pagination.offset / PER_PAGE + 1,
+          perPage: PER_PAGE,
+          sort: [{ by: `name.${locale}`, direction: 'asc' }]
+        })
+        .build();
+      const response = await client.execute({ uri, method: 'GET' });
+      if (response.statusCode === 200) {
+        return {
+          pagination: {
+            count: response.body.count,
+            limit: response.body.limit,
+            total: response.body.total,
+            offset: response.body.offset
+          },
+          products: response.body.results.map(dataTransformer(sdk.parameters.installation))
+        };
+      }
+      throw new Error(response.statusCode);
+    }
+  });
+
+  sdk.window.updateHeight(window.outerHeight);
+}
+
+async function openDialog(sdk, currentValue, config) {
+  const maxFiles = config.maxFiles - currentValue.length;
```

Comment:
why maxFiles?

Reply:
Removed. It was a leftover from DAM apps :hugs: